crates/uv/tests/it/venv.rs

```diff
@@ -491,6 +491,185 @@ fn create_venv_respects_pyproject_requires_python() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn create_venv_respects_group_requires_python() -> Result<()> {
+    let context = TestContext::new_with_versions(&["3.11", "3.9", "3.10", "3.12"]);
+
+    // Without a Python requirement, we use the first on the PATH
+    uv_snapshot!(context.filters(), context.venv(), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Using CPython 3.11.[X] interpreter at: [PYTHON-3.11]
+    Creating virtual environment at: .venv
+    Activate with: source .venv/[BIN]/activate
+    "###
+    );
+
+    // With `requires-python = ">=3.10"` on the default group, we pick 3.10
+    // However non-default groups should not be consulted!
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(indoc! { r#"
+        [project]
+        name = "foo"
+        version = "1.0.0"
+        requires-python = "<3.13"
+        dependencies = []
+        
+        [dependency-groups]
+        dev = ["sortedcontainers"]
+        other = ["sniffio"]
+
+        [tool.uv.dependency-groups]
+        dev = {requires-python = ">=3.10"}
+        other = {requires-python = ">=3.12"}
+        "#
+    })?;
+
+    uv_snapshot!(context.filters(), context.venv(), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Using CPython 3.11.[X] interpreter at: [PYTHON-3.11]
+    Creating virtual environment at: .venv
+    Activate with: source .venv/[BIN]/activate
+    "
+    );
+
+    // When the top-level requires-python and default group requires-python
+    // both apply, their intersection is used. However non-default groups
+    // should not be consulted!
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(indoc! { r#"
+        [project]
+        name = "foo"
+        version = "1.0.0"
+        requires-python = ">=3.11"
+        dependencies = []
+        
+        [dependency-groups]
+        dev = ["sortedcontainers"]
+        other = ["sniffio"]
+
+        [tool.uv.dependency-groups]
+        dev = {requires-python = ">=3.10"}
+        other = {requires-python = ">=3.12"}
+        "#
+    })?;
+
+    uv_snapshot!(context.filters(), context.venv(), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Using CPython 3.11.[X] interpreter at: [PYTHON-3.11]
+    Creating virtual environment at: .venv
+    Activate with: source .venv/[BIN]/activate
+    "
+    );
+
+    // When the top-level requires-python and default group requires-python
+    // both apply, their intersection is used. However non-default groups
+    // should not be consulted!
```

Comment:
How is this intended to be different from the above case?

Reply:
How does it demonstrate anything if the result doesn't change? 🤔

Reply:
`uv venv` aiui completely clobbers the old venv, so each one of these invocations is independent (I've now fixed the issue where 3.11 was first so now these constraints actually do something non-trivial)